ARCs/arc-0041.md

```diff
@@ -0,0 +1,155 @@
+          "desc": "The Application ID of the authority NFT, (0 if ASA)"
+        }
+      ]
+    }
+}
+```
+
+A `arc41_AccountCreated` event SHOULD be emitted, when a vault is created.
+
+
+### Interface
+```json
+{
+  "name": "ARC-41",
+  "desc": "Authority NFT's Vault Base Interface",
+  "methods": [
+    {
+      "name": "arc41_isOwner",
+      "desc": "Indicate if the caller detain the Authority NFT of this vault, this method MUST be called internally by every other methods of the application, it must call ARC-72 ownerOf in case of a application NFT",
+      "readonly": true,
+      "args": [
+        {
+          "type": "uint64",
+          "name": "asaId",
+          "desc": "The asaId owner of the authority NFT, (0 if application NFT)"
+        },
+        {
+          "type": "uint256",
+          "name": "tokenId",
+          "desc": "The ID of the authority NFT, (0 if ASA)"
+        },
+        {
+          "type": "uint64",
+          "name": "AppId",
+          "desc": "The Application ID of the authority NFT, (0 if ASA)"
+        }
+      ]
+      "returns": { "type": "bool", "desc": "True if the caller detain the Authority NFT of this vault, False otherwise" }
+    },
+    {
+      "name": "arc41_optIn",
+      "desc": "Opt-in ASA in the vault",
+      "readonly": false,
+      "args": [
+        { "type": "uint64", "name": "assetId" }
+      ],
+      "returns": { "type": "void" }
+    },
+    {
+      "name": "arc41_transferAlgoFrom",
+      "desc": "Transfers Algos outside of the Vault",
+      "readonly": false,
+      "args": [
+        { "type": "address", "name": "from" },
+        { "type": "address", "name": "to" }
+      ],
+      "returns": { "type": "void" }
+    },
+    {
+      "name": "arc41_transferFrom",
+      "desc": "Transfers ownership of an NFT/fungible Token",
+      "readonly": false,
+      "args": [
+        { "type": "address", "name": "from" },
+        { "type": "address", "name": "to" },
+        { "type": "uint64", "name": "amount" },
+        {
+          "type": "uint64",
+          "name": "asaId",
+          "desc": "The asaId owner of the authority NFT, (0 if application NFT)"
+        },
+        {
+          "type": "uint256",
+          "name": "tokenId",
+          "desc": "The ID of the authority NFT, (0 if ASA)"
+        },
+        {
+          "type": "uint64",
+          "name": "AppId",
+          "desc": "The Application ID of the authority NFT, (0 if ASA)"
+        }
+      ],
+      "returns": { "type": "void" }
+    }
+  ]
+}
+```
+
+The `arc41_isOwner` method MUST be called internaly by every other methods.
+The `arc41_transferAlgoFrom` method MUST error when `from` is not the owner of `asaId` or `tokenId`.
+The `arc41_transferFrom` method MUST error when `from` is not the owner of `asaId` or `tokenId`.
+
+## Rationale
+ASAs NFT ([ARC-3](./arc-0003.md), [ARC-69](./arc-0069.md)) and Application NFT ([ARC-72](./arc-0072.md)) are compatible with This standard.
```

Comment:
Although ARC-3, ARC-19, ARC-69 are all just ASAs and nothing about this spec cares about their initial config or metatadata variations (and just tracks an ASA ID), I'd add ARC-19 to this list for completeness - otherwise some might think it doesn't apply.